extensions/schema-registry/devservice/deployment/src/main/java/io/quarkus/apicurio/registry/devservice/DevServicesApicurioRegistryProcessor.java

```diff
@@ -259,9 +259,6 @@ private ApicurioRegistryContainer(DockerImageName dockerImageName, int fixedExpo
                 withLabel(DEV_SERVICE_LABEL, serviceName);
             }
             withEnv("QUARKUS_PROFILE", "prod");
-            if (!dockerImageName.getRepository().equals("apicurio/apicurio-registry-mem")) {
```

Comment:
If

```
quarkus.apicurio-registry.devservices.image-name=quay.io/apicurio/apicurio-registry-mem:2.2.1.Final
```

fails this check (as described in #24308), then there's a bug somewhere. This condition was supposed to verify the image repository name, regardless of the registry name.

I suppose your "fork" of the Apicurio Registry image in an internal registry also lives in the `apicurio/apicurio-registry-mem` repository.

(Note I'm using the terms registry and repository in their very Docker-centric meaning. That is, in `quay.io/apicurio/apicurio-registry-mem:2.2.1.Final`, the `quay.io` part is a _registry_, `apicurio/apicurio-registry-mem` is a _repository`, and `2.2.1.Final` is a _tag_.)

If possible, I'd like to keep this check, but make it work as intended :-)

Reply:
@Ladicek Thanks for the accurate explanation. Now I understand what our problem actually ist. 
The 3rd Party images contained in our registry are under a specific directory, in other words, our apicurio images look something like `our-registry.com/dir/apicurio/apicurio-registy-mem`. 

For these images the check above cause the exception described at https://github.com/quarkusio/quarkus/issues/24308

I totally see your point of not removing the check, but may be it only should be checked if the registry  ends with `apicurio/apicurio-registry-mem`

Reply:
Hmm yea, namespacing like this is probably quite common, I guess. I guess what we should check is: the repository is equal to `apicurio/apicurio-registry-mem` || the repository ends with `/apicurio/apicurio-registry-mem`.

Reply:
But just checking it it ends with `apicurio/apicurio-registry-mem`, or even just _contains_ `apicurio/apicurio-registry-mem`, would probably be enough. We can't truly prevent people from shooting themselves in the foot :-)

Reply:
I think, just checking if the registry ends with the expected apicurio image's name should be enough to prevent starting the DevService with a wrong or unexpected image.
New version is pushed 👍

Reply:
Open question but I wonder if we should make this a warning instead? The current check seems to be very restrictive and we could imagine someone having its own image based on `apicurio/apicurio-registry-mem`.
Just a thought, I'll let @Ladicek and @ozangunalp decide.

Reply:
When we did this dev service, other dev services had the same check (I think we mainly looked at Kafka). Not sure what's the current state, but I'll take a look.

Reply:
Kafka dev service is more permissive now. If the image name contains strimzi we treat it as a strimzi image, otherwise as a Redpanda image.

Reply:
The Apicurio Registry dev service config javadoc says _Note that only Apicurio Registry 2.x images are supported_.

To compare, the AMQP (ActiveMQ Artemis) dev service config javadoc says _Note that only {@code quay.io/artemiscloud/activemq-artemis-broker} images are supported_ and the RabbitMQ dev service config javadoc doesn't say anything about a similar requirement, even though it enforces the docker image repository to be equal to `rabbitmq`.

So we're not exactly consistent here :-)